src/api/goods.ts

```diff
@@ -5,15 +5,41 @@ const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/good
 
 export function getAll(): Promise<Good[]> {
   return fetch(API_URL)
-    .then(response => response.json());
+    .then(response => {
+      if (!response.ok) {
+        // eslint-disable-next-line no-console
+        console.error('Cannot fetch goods');
+
+        return [];
+      }
+
+      return response.json();
+    });
 }
 
 export const get5First = () => {
   return getAll()
-    .then(goods => goods); // sort and get the first 5
+    .then(goods => {
+      const sortedGoods = goods.sort((a, b) => a.name.localeCompare(b.name));
+      const first5Goods = sortedGoods.slice(0, 5);
+
+      return first5Goods;
+    })
+    .catch(error => {
+      // eslint-disable-next-line no-console
+      console.error('Cannot fetch 5 first goods', error);
+
+      return [];
+    });
```

Comment:
when you return a initial value(empty array) in getAll you can remove this code from `get5First` and `getRedGoods`

Reply:
@dffUqp thanks, it's important experience for me